src/app/layout/redirect-status/redirect-status.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { async, TestBed } from '@angular/core/testing';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { RedirectStatusComponent } from './redirect-status.component';
+
+describe('Redirect Status Component', () => {
+  let fixture;
+  let routeMock: any;
+
+  beforeEach(() => {
+    routeMock = {
+      snapshot: {
+        params: {
+          redirectType: '_verifyEmail'
+        },
+        queryParams: {
+          status: 'fail',
+          error: 'Some error'
+        }
+      }
+    };
+
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule.withRoutes([])],
+      declarations: [RedirectStatusComponent],
+      providers: [
+        {
+          provide: ActivatedRoute, useValue: routeMock
+        }
+      ],
+      // Tells the compiler not to error on unknown elements and attributes
+      schemas: [NO_ERRORS_SCHEMA]
+    });
+
+    fixture = TestBed.createComponent(RedirectStatusComponent);
+  });
+
+  it('should verify redirect status and error message is set through activated route', async(() => {
+    let comp = fixture.componentInstance;
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
```

Comment:
Is the `async` and `whenStable` really necessary here? Also, this spec could be cleaned up to use the `TestContext` and `initContext` testing utils we have in the project, which might reduce a bit of boilerplate but more importantly make it the same consistent boilerplate as we see in many other places. IMO it's more readable and maintainable that way. I think it'll also help in removing the async/whenStable stuff since it has a little bit of included tooling around the component lifecycle.